### ISSUE TITLE
Rendering bulgarian texts ends up as UTF-16 escaped

### ISSUE THREAD

**Issue by ptahchiev** [thymeleaf/thymeleaf#681]
Hello,

I'm using thymeleaf 3.0.9.RELEASE and I'm rendering the following Bulgarian text: "Безплатна доставка на следващия ден" and it ends up rendered as \u0411\u0435\u0437\u043F\u043B\u0430\u0442\u043D\u0430 \u0434\u043E\u0441\u0442\u0430\u0432\u043A\u0430 \u043D\u0430 \u0441\u043B\u0435\u0434\u0432\u0430\u0449\u0438\u044F \u0434\u0435\u043D

Here's my method:

````
    @Override
    public String parseContent(@Nonnull final String content, final Map<String, Object> params) {
        Assert.notNull(content, "Parameter content must be non-null");
        return templateEngine.process("<span th:inline=\"javascript\">" + content + "</span>", createSpringWebContext(params));
    }
```` 

Where content is 

"Купете стоки на стойност повече от [[${currencyFormatter.print(promotion.activeSubtotalThreshold, locale)}]] за да получите [[${promotion.quantity}]] of [[${promotion.product.getName(locale)}]] БЕЗПЛАТНО!"

and promotion is a variable in the context with name in Bulgarian locale being "Безплатна доставка на следващия ден". 

I found out the problem is because when you are creating your `JacksonStandardJavaScriptSerializer` you are specifying this:

````
this.mapper.enable(JsonGenerator.Feature.ESCAPE_NON_ASCII);
````

making it like this:

````
this.mapper.disable(JsonGenerator.Feature.ESCAPE_NON_ASCII);
````

works perfectly fine. I will submit a PR.

**Comment by ptahchiev** [comment]
Here's the PR https://github.com/thymeleaf/thymeleaf/pull/682

**Comment by danielfernandez** [comment]
Thanks. Could you please clarify the reason why the escaping isn't correct? From my tests, the escape sequence does correctly correspond to the text you specify in Bulgarian characters..

**Comment by danielfernandez** [comment]
Also, please note that you are rendering JavaScript-treated text (and therefore JavaScript-escaped text) inside an HTML `<span>` element... is this what you intended?

**Comment by ptahchiev** [comment]
Hi Daniel,

the bulgarian text: 
```
""Купете стоки на стойност повече от [[${currencyFormatter.print(promotion.activeSubtotalThreshold, locale)}]] за да получите [[${promotion.quantity}]] от [[${promotion.product.getName(locale)}]] БЕЗПЛАТНО!""
```
means 
```
"Buy products for more than [[${currencyFormatter.print(promotion.activeSubtotalThreshold, locale)}]] to get [[${promotion.quantity}]] of [[${promotion.product.getName(locale)}]] FREE!"
```
what I would like to achieve is render the final output: 
````
"Buy products for more than 100$ to get 1 of "Africa Love Bandeau Bikini" FREE" 
````
in Bulgarian. But instead what I get is the product name rendered as UTF-16 escaped so half of the message is good, and half of it is escaped:
````
"Купете стоки на стойност повече от 100$ за да получите 1 от "\u0411\u0435\u0437\u043F\u043B\u0430\u0442\u043D\u0430 \u0434\u043E\u0441\u0442\u0430\u0432\u043A\u0430 \u043D\u0430 \u0441\u043B\u0435\u0434\u0432\u0430\u0449\u0438\u044F \u0434\u0435\u043D"  БЕЗПЛАТНО!
````
That's why I think the escaping is not correct.

**Comment by danielfernandez** [comment]
OK, but what I mean is that apparently you are rendering that text inside an HTML element (`<span>`), but you are using `th:inline="javascript"`, which considers what is processed inside as JavaScript code and therefore applies Jackson and, in your case, JavaScript-escaping. Thus the unicode escapes, which are perfectly valid JavaScript escaped text.

So I'm confused about your use of `th:inline="javascript"` in a `<span>` element. If you are doing that only because you want your `[[${...}]]` expressions resolved, and seeing you are using Thymeleaf 3.0, you don't really have to use inlining at all.

Assuming you are using the `StringTemplateResolver`, this:
```java
return templateEngine.process("<span>" + content + "</span>", createSpringWebContext(params));
```
or even simply this:
```java
return templateEngine.process(content, createSpringWebContext(params));
```
...should work for you. There will be escaping anyway for your Bulgarian characters, but it will be HTML escaping and not JavaScript, which I presume is what you are looking for.

**Comment by ptahchiev** [comment]
Hi @danielfernandez 

thank you for the response - it works perfectly fine with no javascript inlining. I guess I was confused because this used to work with thymeleaf 2.x

Once again, thank you for your fast response.
